inst/private/python_ipc_popen2.m

```diff
@@ -43,11 +44,29 @@
   info = [];
 
   if (strcmp(what, 'reset'))
-    cleanup = [];  % triggers a call to reset function
+    if (~ isempty (pid))
+      if (verbose)
+        disp ('Closing the Python communications link.')
+      end
+    end
+    A = true;
+    if (~ isempty (fin))
+      % produces a single newline char: not sure why
+      t = fclose (fin);
+      fin = [];
+      waitpid (pid);
+      pid = [];
+      A = A && (t == 0);
+    end
+    if (~ isempty (fout))
+      t = fclose (fout);
+      fout = [];
+      A = A && (t == 0);
+    end
+    emergency_cleanup = [];  % note: triggers emergency file-close
```

Comment:
I'm confused about this line. Why do we need to "trigger emergency file-close"? IIUC we've just cleaned things up above.

Reply:
Its going to get triggered eventually.  Might as well trigger it now, when I know it is an no-op.

I guess I'm vague uncomfortable with this split: here on line 63 we close `fout`, let's say its 24.  My understanding of how anonymous functions work is that the onCleanup still has "24" hardcoded in it (it has no idea `fout` has been replaces with `[]`).

Some other `fopen` call later reuses 24.

Now our ticking timebomb onCleanup goes off and closes that 24.  I don't know if that can really happen but the idea that it might means I don't like that onCleanup lying around.   Let it go off right now while its harmless.

Reply:
maybe I should have a comment that gives that idea: `% trigger while we know it is a no-op`?

Reply:
Sure, this should make things clearer